### PR TITLE
 task 5137 create ui overall genres

### PR DESCRIPTION
**Pull request by QNFC**
Create UI overall genres
Link redmine : https://edu-redmine.framgia.vn/issues/5137
genresoverall

### CODE REVIEW

File: app/src/main/java/com/framgia/music_48/data/model/Genres.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package com.framgia.music_48.data.model;
+
+public class Genres {
+    private String mImageGenres;
+
+    public Genres(String imageGenres) {
+
```
Comment (QuangHieuu) [inline app/src/main/java/com/framgia/music_48/data/model/Genres.java]: remove blank

File: app/src/main/java/com/framgia/music_48/screen/Genres/Adapter/GenresAdapter.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import android.widget.ImageView;
+import com.framgia.music_48.R;
+import com.framgia.music_48.data.model.Genres;
+import java.util.List;
+
+public class GenresAdapter extends RecyclerView.Adapter<GenresAdapter.GenresHolder> {
+    private List<Genres> mGenres;
+    private Context mContext;
+
+    public GenresAdapter(List<Genres> genres, Context context) {
```
Comment (QuangHieuu) [inline app/src/main/java/com/framgia/music_48/screen/Genres/Adapter/GenresAdapter.java]: context luôn được truyền vào đầu tiên

File: app/src/main/res/layout/fragment_genres.xml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    android:background="@color/color_grey_500"
+    >
+
+    <android.support.v7.widget.RecyclerView
+        android:id="@+id/recyclerViewGenres"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        app:layoutManager="android.support.v7.widget.LinearLayoutManager"
+        />
+
```
Comment (QuangHieuu) [inline app/src/main/res/layout/fragment_genres.xml]: remove blank

File: app/src/main/res/layout/item_genres.xml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+    <android.support.constraint.ConstraintLayout
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        >
+
+        <ImageView
+            android:id="@+id/imageGenres"
+            android:layout_width="wrap_content"
+            android:layout_height="150dp"
```
Comment (QuangHieuu) [inline app/src/main/res/layout/item_genres.xml]: hard code

File: app/src/main/res/layout/item_genres.xml [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+            />
+    </android.support.constraint.ConstraintLayout>
+</android.support.v7.widget.CardView>
```
Comment (QuangHieuu) [inline app/src/main/res/layout/item_genres.xml]: add blank

**Review by QuangHieuu — APPROVED** [review]

**Comment by QNFC** [comment]
ready

File: app/src/main/java/com/framgia/music_48/screen/Genres/Adapter/GenresAdapter.java [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+import com.framgia.music_48.data.model.Genres;
+import java.util.List;
+
+public class GenresAdapter extends RecyclerView.Adapter<GenresAdapter.GenresHolder> {
+    private List<Genres> mGenres;
+    private Context mContext;
+
+    public GenresAdapter(Context context, List<Genres> genres) {
+        mGenres = genres;
+        mContext = context;
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/framgia/music_48/screen/Genres/Adapter/GenresAdapter.java:19]: context luôn luôn đứng ở vị trí đầu e

**Comment by QNFC** [comment]
ready

File: app/src/main/java/com/framgia/music_48/screen/Genres/Adapter/GenresAdapter.java [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+    public void onBindViewHolder(@NonNull GenresHolder genresHolder, int i) {
+        genresHolder.bindData(mGenres.get(i));
+    }
+
+    @Override
+    public int getItemCount() {
+        return mGenres != null ? mGenres.size() : 0;
+    }
+
+    class GenresHolder extends RecyclerView.ViewHolder {
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/framgia/music_48/screen/Genres/Adapter/GenresAdapter.java]: https://stackoverflow.com/questions/13516835/why-in-viewholder-pattern-should-the-viewholder-class-be-static

**Comment by QNFC** [comment]
ready

**Review by tuanlvt-1117 — APPROVED** [review]

**Pull request merged**